node/modules/storageminer.go

```diff
@@ -512,33 +516,35 @@ func F3Participation(mctx helpers.MetricsCtx, lc fx.Lifecycle, node v1api.FullNo
 	)
 	ctx, cancel := context.WithCancel(mctx)
 	var wg sync.WaitGroup
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
-
-		participator := newF3Participator(
-			node,
-			participant,
-			&backoff.Backoff{
-				Min:    1 * time.Second,
-				Max:    1 * time.Minute,
-				Factor: 1.5,
-			},
-			checkProgressMaxAttempts,
-			checkProgressInterval,
-			leaseTerm,
-		)
-
-		switch err := participator.participate(ctx); {
-		case err == nil, errors.Is(err, context.Canceled):
-			log.Infof("Stopped participating in F3")
-		default:
-			log.Errorw("F3 participation stopped abruptly", "err", err)
-		}
-	}()
-
 	lc.Append(fx.Hook{
-		OnStop: func(ctx context.Context) error {
+		OnStart: func(context.Context) error {
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+
+				participator := newF3Participator(
+					node,
+					participant,
+					&backoff.Backoff{
+						Min:    1 * time.Second,
+						Max:    1 * time.Minute,
+						Factor: 1.5,
+					},
+					checkProgressMaxAttempts,
+					checkProgressInterval,
+					leaseTerm,
+				)
+
+				switch err := participator.participate(ctx); {
+				case err == nil, errors.Is(err, context.Canceled):
```

Comment:
nit: I generally avoid looking at the error itself and instead check `ctx.Err` and/or `ctx.Done`.

1. Services will occasionally fail, e.g., because some internal context gets canceled (shouldn't happen, but does).
2. Canceling the context can sometimes cause other follow-on errors which should generally be ignored (although I guess we do want to, e.g., learn about shutdown errors if they involve failing to record state, etc.).